tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts

```diff
@@ -149,12 +149,12 @@ export class AnnotationComponent implements AfterViewInit, OnChanges {
     this.data.forEach((element) => {
       runs.add(element.run);
     });
-    this.runs = [...runs];
-    this.svg.style('height', `${this.runs.length * this.runHeight}px`);
-    this.height = this.runs.length * this.runHeight;
+    this.runs = this.activeRuns.filter((run) => runs.has(run));
+    this.svg.style('height', `${this.activeRuns.length * this.runHeight}px`);
     this.width = this.annotationContainer.nativeElement.clientWidth || 10;
     this.chartWidth = this.width - this.margin.left - this.margin.right;
-    this.chartHeight = this.height - this.margin.top - this.margin.bottom;
+    this.chartHeight =
+      this.runs.length * this.runHeight - this.margin.top - this.margin.bottom;
```

Comment:
Is it intentional that `this.svg.style.height` is out of sync with `this.chartHeight`?  It's not clear to me why the former uses `activeRuns.length` and the latter uses `runs.length`.  Doesn't that lead to cases where the chartHeight is clipped?

Reply:
The SVG should always be `activeRuns.length * runHeight` to ensure all rows have the same height. I need to do this to be able to make use of `cdk-virtual-scroll-viewport`.

The drawing area which also determines the scales and thus y-positions of all d3 elements should, however, only consider the runs that the current annotation contains.

I added code comments for that.